hw/ip/flash_ctrl/data/flash_ctrl.sv.tpl

```diff
@@ -279,20 +301,29 @@ module flash_ctrl import flash_ctrl_pkg::*; (
   // hardware interface
 
   // software only has privilege to change creator seed when provision enable is set and
-  // before the the seed is set as valid in otp
-  assign creator_seed_priv = lc_provision_en[FlashLcCreatorSeedPriv] == lc_ctrl_pkg::On &
+  // before the the seed is set as valid in otp.
+  // lc provision write enable is used here as creator assets can only be changed when
+  // creator secrets are not yet locked.
+  assign creator_seed_priv = (lc_provision_wr_en[FlashWrLcCreatorSeedPriv] == lc_ctrl_pkg::On) &
                              ~otp_i.seed_valid;
 
   // owner seed is under software control and can be modided whenever provision enable is set
-  assign owner_seed_priv = lc_provision_en[FlashLcOwnerSeedPriv] == lc_ctrl_pkg::On;
+  // read enable is used here as this is mostly under the control of creator software and just
+  // needs to be locked out from specific life cycle states.
+  assign owner_seed_priv = lc_provision_rd_en[FlashRdLcOwnerSeedPriv] == lc_ctrl_pkg::On;
+
+  // the seed is only readable after it has been written and locked.
+  logic seed_rd_en;
+  assign seed_rd_en = lc_provision_wr_en[FlashWrLcMgrIf] == lc_ctrl_pkg::Off &
+                      lc_provision_rd_en[FlashRdLcMgrIf] == lc_ctrl_pkg::On;
```

Comment:
but if RD is the complement of WR, wouldn't that mean from the beginning of time (ie raw), WR would be 0 and RD would be 1? and then seed_rd_en would become 1? Am I understanding you correctly?

Good point about the sync differences... it might not matter.. because the flash logic that reads this is kicked off after the `pwrmgr` receives the indication that life cycle is complete.

But still, let me know if i understood you right, because if this can be done right by construction, that would be even better.  If it's better, I also don't mind changing it on the LC side to make a provision signal specific for creator, another specific for owner.

Reply:
actually look at this more. I don't think this is quite right. 
Let me enumerate here in a bit what I need, and we can work out what's right signals to use.

Reply:
Software Access Table 
| Asset                  |   Raw / Test Locked   |  Test unlocked         | Production states  | RMA |
|------------------      | ----------------------------  | -------------------         | ------------------------ |  ----- | 
| creator seed       |  no access                  |  no access               | accessible if ID blank | accessible |
| owner seed        |   no access                 |  no access               | accessible  | accessible |
| isolated partition |  no access                   |  writeable               | accessible |  accessible  |

Hardware Access Table 
| Asset                  |   Raw / Test Locked   |  Test unlocked         | Production states / RMA |
|------------------      | ----------------------------  | -------------------         | ------------------------ | 
| creator seed       |  no access                  |  no access               | accessible if ID not blank | 
| owner seed        |   no access                 |  no access               | accessible if ID not blank |
| isolated partition |  no access                  |  no access               | no access                       |

Reply:
okay i should put this table somewhere.
Let me know what you think.

Reply:
so what I meant by complement was to make these signals complements only during the PROD, PROD_END, DEV and RMA states. otherwise both would be OFF. 
---------------
RE the tables:

for the HW side, we could use the redefined `provision_rd_en` that is the complement of `provision_wr_en` for controlling access to the seeds.

for the SW side, I have a question: does the creator seed have to be `accessible` in RMA instead of `accessible if ID blank`? if we can align that, then we could do the following allocation:

- use `provision_wr_en` for the creator seed (RW)
- use `provision_rd_en` || `provision_wr_en` for the owner seed (RW)
- use `provision_rd_en` || `provision_wr_en` for the isolated partition (R) 
- use `provision_rd_en` || `provision_wr_en` || `nvm_debug_en` for the isolated partition (W)  

if that does not work, I believe creating separate signals might be a cleaner route (then we can also give these signals more distinctive names. e.g.,
`creator_seed_sw_rw_en`
`owner_seed_sw_rw_en`
`iso_part_sw_rd_en`
`iso_part_sw_wr_en`
`seed_hw_rd_en`

I can add/expand the LC documentation once we've fixed this.

Reply:
But where would you do this complement though? Wouldn't you need to do it in life cycle controller..? Because if i were to do it in the destination module, I would also need the state information right? And that's not something we're broadcasting out?

I do think for RMA, we want to allow as much visibility as possible. Because the device "may" have screwed up due to a bit flips in the flash, and it may be reasonable for whatever debugging to look at that.

Reply:
anyways, i'm tempted to go for your second proposal because i feel like the boundaries will be cleaner.
That way, we don't have to assume i'm constructing the logic a certain way on the `flash` side, and if ever the properties of the provisioning signals change based on the states, we won't need to re-open this discussion to see how things should be re-wired.

The dedicated way seems to make the intent much clearer, and if we ever make modifications on the lc side it will be immediately clear why and what we need to look at.

let me know what you think.  We should also decide where to put that table I enumerated. I could put it in flash and maybe lc could point to it, or vice versa.

Reply:
@msfschaffner does it make sense for me to merge this PR first and we can work out the enables separately? Or do you prefer I keep this in place until that part is settled?

Both are okay with me.

Reply:
Sorry, a bit behind with reviews. 

So the inversion I was talking about would be done in the life cycle controller itself - it's basically a redefinition of when `provision_rd_en` would be asserted. 
Inverting within the destination module feels a bit hacky to me...

SGTM, let's proceed with this PR and amend afterwards. 
I do also lean towards the cleanly separated solution, since that basically simplifies the consumers and pushes all the condition logic into the LC controller.

Reply:
RE location for the table... the life cycle controller spec feels like the more natural place for me since all the control signals get explained there, and we already have a flash collateral section. 

Let's amend the spec once we have made the RTL changes.

Reply:
yeah sounds good. Then let's go with the the separate signals, i feel like that's less error prone in the long haul. 
do you want me to push a PR based on what we discussed for the signal decode side?

Reply:
that would be great! I can take over the documentation part and pile it onto my doc updates in https://github.com/lowRISC/opentitan/pull/4386